migrate/json_to_terraform.py

```diff
@@ -0,0 +1,340 @@
+
+        def print_users(*args):
+            print(*args, file=users_file)
+
+        existing_user_labels = get_existing_user_labels()
+
+        print_users(
+            "// Users who are members or outside collaborators of Open edX organization.",
+        )
+        print_users()
+        print_users(export_comment)
+        for username, user_label in sorted_username_label_pairs:
+            if user_label in existing_user_labels:
+                continue
+            print_users()
+            print_users(f'module "{user_label}" {{')
+            print_users('source = "./modules/user"')
+            print_users(f'username = "{username}"')
+            print_users("}")
+
+    # Finally, turn the roles into terraform 'role' module invocations.
+    # Don't bother indenting nicely; that's what `terraform fmt` is for.
+    with open(output_roles_fpath, "w") as roles_file:
+
+        def print_roles(*args):
+            print(*args, file=roles_file)
+
+        print_roles("// Roles, which grant an (ideally logical) collection of")
+        print_roles(
+            "// repository permissions to one or more Open edX organization users."
+        )
+        print_roles()
+        print_roles(export_comment)
+
+        for role in sorted(roles_to_create, key=lambda role: role.label):
+
+            # Module block header.
+            print_roles()
+            print_roles(f'module "{role.label}" {{')
+            print_roles('source = "./modules/role"')
+
+            # One list of repos for each access level. Example:
+            #   admin_repos = [
+            #   module.repo_edx_platform,
+            #   module.repo_xblock,
+            #   ]
+            #   write_repos = [
+            #   module.repo_frontend_platform,
+            #   ]
+            for access_level in [REPO_ADMIN, REPO_MAINTAIN, REPO_WRITE, REPO_READ]:
+                access_string = access_level_to_string(access_level)
+                repo_labels_for_access_level = sorted(
+                    [
+                        repo_name_to_repo_label(repo_name)
+                        for repo_name, repo_access_level in role.repo_access.items()
+                        if repo_access_level == access_level
+                    ]
+                )
+                if repo_labels_for_access_level:
+                    print_roles(f"{access_string}_repos = [")
+                    for repo_label in repo_labels_for_access_level:
+                        print_roles(f"module.{repo_label},")
+                    print_roles("]")
+
+            # Enumerate role members, alphabetized,
+            print_roles("users = [")
+            for member in sorted(member.lower() for member in role.members):
+                member_label: UserLabel = username_to_user_label(member)
+                print_roles(f"module.{member_label},")
+            print_roles("]")
+
+            # Closing brace for module block.
+            print_roles("}")
```

Comment:
Nit: Some of these "output a file with this stuff in it" pieces could be broken out into separate functions.

Reply:
Agreed -- I was constantly fighting the urge to refactor these scripts.